src/crowdin_client.cpp

```diff
@@ -50,64 +57,71 @@
     using std::smatch;
 #endif
 
+
+
 // ----------------------------------------------------------------
 // Implementation
 // ----------------------------------------------------------------
 
 namespace
 {
 
-#define OAUTH_CLIENT_ID     "poedit-y1aBMmDbm3s164dtJ4Ur150e2"
-#define OAUTH_AUTHORIZE_URL "/oauth2/authorize?response_type=token&client_id=" OAUTH_CLIENT_ID
+#define OAUTH_SCOPE         "project"
+#define OAUTH_CLIENT_ID     "k0uFz5HYQh0VzWgZmOpA"
+//      any arbitrary unique unguessable string (e.g. UUID in hex)
+#define OAUTH_STATE         "948cf13ffffb47119d6cfa2b68898f67"
+//      The value of below macro should be set exactly as is (without quotes)
+//      to "Authorization Callback URL" of Crowdin application
+//      https://support.crowdin.com/enterprise/creating-oauth-app/
 #define OAUTH_URI_PREFIX    "poedit://auth/crowdin/"
-
-// Recursive extract files from /api/project/*/info response
-void ExtractFilesFromInfo(std::vector<std::wstring>& out, const json& r, const std::wstring& prefix)
-{
-    for (auto i : r["files"])
-    {
-        std::wstring name = prefix + str::to_wstring(i["name"]);
-        std::string node_type = i["node_type"];
-        if (node_type == "file")
-        {
-            out.push_back(name);
-        }
-        else if (node_type == "directory" || node_type == "branch")
-        {
-            ExtractFilesFromInfo(out, i, name + L"/");
-        }
-    }
-}
+#define OAUTH_AUTHORIZE_URL "/oauth/authorize" \
+    "?" "response_type" "=" "token" \
+    "&" "scope"         "=" OAUTH_SCOPE \
+    "&" "client_id"     "=" OAUTH_CLIENT_ID \
+    "&" "state"         "=" OAUTH_STATE \
+    "&" "redirect_uri"  "=" OAUTH_URI_PREFIX \
+    "&" "utm_source=poedit.net&utm_medium=referral&utm_campaign=poedit" // source tracking
 
 } // anonymous namespace
 
-
 std::string CrowdinClient::WrapLink(const std::string& page)
 {
-    std::string url("https://poedit.net/crowdin");
+    std::string url("https://accounts.crowdin.com");
     if (!page.empty() && page != "/")
-        url += "?u=" + http_client::url_encode(page);
+        url += page;
     return url;
 }
```

Comment:
Addressed 122ae7f . Also rearranged/refactored OAuth URL related code to get rid of currently needless overcomplications with `WrapLink()` since now we don't use redirect via [poedit.net](htps://poedit.net) or anything else that really needs to `WrapLink()` resulting compounding of URL from different parts from different places of code...unless I miss something so please correct me if otherwise. But honestly speaking I really experienced some  complications when tried to follow `WrapLink()` first time (it was one of my first acquaintance with Poedit source code) which I found unjustified for so simple purpose or use case. Absolutely no blame on you but just recalling your care for anybody new who might start to work on Poedit in the future, so here is humble opinion and feedback of such person here :) (since that time it was pretty new for me :) ).

Reply:
`WrapLink` is certainly an unfortunate name and the comment doesn't do a good job explaining it; I should renamed it to something like `CrowdinClient::CreateLinkToPage`.

But that doesn't mean it is not needed, and if anything, your PR shows the need for it: when changing the way Crowdin is linked, it is better to do it in one place, not in several places; in that regard, hardcoding UTM tracking is a regression. But not worries, I'll address it myself.

Reply:
Could you please look how it's addressed now in rebased and force-pushed above mentioned commit a046666ddfea90b2dec190b3a204a12e4fc699f6 in response on your below messages (which disappeared from GitHub after rebasing and force-pushed above commit)
> Doing this successfully broke the "Learn more about Crowdin" link — while it previously went to crowdin.com, a page where one can learn what Crowdin is and what it has to offer, it now goes directly to a nondescript login page.

> Let's just restore WrapLink to how it was before this PR and call it a day?

I would highly recommend against this due of reasons I already mentioned in my previous message by this conversation. Plus in response to your more resent messages I would not mix usage of same function call it `CrowdinClient::CreateLinkToPage()` or whatever else since now base URL for OAuth and used to "Learn more about Crowdin" are different now, i.e. pointing on different unrelated resources which just lead to even more confusion when using single function for them. Also sorry if I missed some of important points in your explanation but I still don't get necessity of such single common function now when there are clearly separated and quite straightforward flow/logic/code/place for generation of URL for OAuth and absolutely unrelated link for URL "Learn more about Crowdin". 

>  it is better to do it in one place, not in several places; in that regard

But currently all of them are in `CrowdinClient` (`crowdin_client.h/cpp`) very closely each from other in the code (2 lines line by line in [`crowdin_client.h`](https://github.com/tetatetit/poedit/blob/a6788fa535fefc63a8957424a8dfb50a6796d588/src/crowdin_client.h#L51) and just in 2 logical places in `crowdin_client.cpp`  - [one at the beginning of file for macros](https://github.com/tetatetit/poedit/blob/a6788fa535fefc63a8957424a8dfb50a6796d588/src/crowdin_client.cpp#L72) and [another in `CrowdinClient::Authenticate()` ](https://github.com/tetatetit/poedit/blob/a6788fa535fefc63a8957424a8dfb50a6796d588/src/crowdin_client.cpp#L141)

> in that regard, hardcoding UTM tracking is a regression. But not worries, I'll address it myself.

Thanks a lot for taking care for this, but I am still not sure I understand what you mean here so I would higly appreciate if you would clarify (just to get what I understand wrong to learn lesson for future or so). I have put UTM to the separate macro in above 1st place of code and use it consistently in the 2nd place above. So what is wrong I did with it?